src/providers/database/FirebaseClient.ts

```diff
@@ -308,7 +316,10 @@ export class FirebaseClient implements IFirebaseClient {
     if (!hasRawFile) {
       return;
     }
-    ref.src = await this.uploadAndGetLink(ref.rawFile, docPath, fieldPath);
+
+    const sourceFieldName = Object.keys(ref).find((key) => key !== 'rawFile' && key !== 'title');
+
+    ref[sourceFieldName] = await this.uploadAndGetLink(ref.rawFile, docPath, fieldPath);
     delete ref.rawFile;
   }
 
```

Comment:
I image the `uploadAndGetLink` function also needs to reflect the new object tree.